model/task_lifecycle.go

```diff
@@ -1643,26 +1643,57 @@ func ClearAndResetStrandedHostTask(h *host.Host) error {
 	return nil
 }
 
-// ResetStaleTask fixes a task that has exceeded the heartbeat timeout by
-// marking the current task execution as finished and, if possible, a new
-// execution is created to restart the task.
+// ResetStaleTask fixes a task that has either exceeded the heartbeat timeout
+// or has been marked as aborted but was never ended by the agent.
+// The current task execution is marked as finished and, if the task is not
+// aborted, a new execution is created to restart the task.
+// If the task is aborted, the task is marked as failed alongside necessary
+// logic that would typically be run by the agent.
```

Comment:
Instead of saying this is specifically on the agent, can we phrase this as something along the lines of "the task is marked as failed and any other updates to finish the task are done".